### DISCUSSION TITLE
Can I avoid the use of the built-in dependency notation?

### DISCUSSION THREAD

**Discussion by podollb** [Splitties/refreshVersions#701] [Q&A]
Is there a way to avoid the use of the **built-in dependency notation**? If I have existing `versions.properties` files and/or new projects that I want to migrate to `refreshVersions`, but I do **_NOT_** want to use the **built-in dependency notation**, is there a flag I can set to do that?

https://splitties.github.io/refreshVersions/dependency-notations/

**Comment by LouisCAD** [comment]
Yep, just use `_` in place of the version, as all documented on this page: https://splitties.github.io/refreshVersions/add-dependencies/

**Reply by podollb** [reply]
Just to be clear, when I migrate a project to use `refreshVersions` it generates/replaces some of the dependencies with the **built-in dependency notation**, like this for example, `implementation(Kotlin.stdlib.jdk8)`, but I do not want it to do that, I would rather have it always use the actual notation `implementation("org.jetbrains.kotlin:kotlin-stdlib-jdk8:_")`, so I'm wondering about a way to turn-off auto-generation of the built-in notation.

**Reply by LouisCAD** [reply]
Why don't you want the built-in dependency notations?

The reasons we made them is because there are a lot of repeated words in maven-style dependencies, risk of typos, made worse with repetitive words, and we can also bring KDoc that links to the release notes and other helpful info.

**Reply by podollb** [reply]
We have a lot of automation, and using standard notation in the build.gradle(.kts) files makes a lot of that easier, rather than having to know the built-in notation keys and cross referencing those with the actual dependency notation (i.e. if a tool/script scans build.gradle(.kts) files, across many git orgs/repos, we'd prefer not to have to know about, and/or lookup the special notation).

**Reply by LouisCAD** [reply]
You are not scanning dependencies added by Gradle plugins with this technique, FYI.

Can you give a bit more information about the kind of automation that requires this data, the use case you have?

**Reply by podollb** [reply]
The kind of automation, at the basic level, is simple text parsing of build.gradle.kts files.

**Reply by LouisCAD** [reply]
That's not telling why you are parsing those files.

I'm asking because there might be better solutions than what you're asking for.
For example, there's this Gradle plugin that suits several dependency tracking related use cases.

**Reply by podollb** [reply]
Sorry for being vague, but there are a few types of automation tools that parse the build.gradle.kts files (across many git organizations and repositories) to glean information (for the most part, it's simply being treated, and seen, as a text file -- gradle is not involved).

**Reply by LouisCAD** [reply]
Forgot to link the Gradle plugin I was referring to: https://github.com/dropbox/dependency-guard
